src/Models/BlogModel.php

```diff
@@ -47,6 +64,20 @@ public function getBlogById(int $id): Blog
 
     public function hydrateSingleBlog(array $data): Blog
     {
-        return new Blog($data['id'], $data['title'], $data['content'], $data['authorid'], $data['posttime']);
+        if ($data['username'] == null) {
+            return new Blog($data['id'], $data['title'], $data['content'], $data['authorid'], 'Anonymous', $data['posttime']);
+        } else {
+            return new Blog($data['id'], $data['title'], $data['content'], $data['authorid'], $data['username'], $data['posttime']);
+        }
```

Comment:
Rather than repeating the instantiation code, you should just set a $username variable in this if to be either anon or $data['username'] and then return the new Blog after

Reply:
try this:

$username =  $data['username']  ?: Anonymous;
return new Blog($data['id'], $data['title'], $data['content'], $data['authorid'], $username, $data['posttime']);